api/src/main/java/org/apache/iceberg/stats/StatsUtil.java

```diff
@@ -0,0 +1,172 @@
+  private StatsUtil() {}
+
+  public static int statsFieldIdFor(int fieldId) {
+    int idSpaceStart = DATA_SPACE_FIELD_ID_START;
+    int id = fieldId;
+    if (fieldId >= RESERVED_FIELD_IDS_START) {
+      // this is a reserved field ID, which uses a different calculation
+      idSpaceStart = METADATA_SPACE_FIELD_ID_START;
+      id = RESERVED_FIELD_IDS - (Integer.MAX_VALUE - fieldId);
+    }
+
+    long finalId = idSpaceStart + NUM_STATS_PER_COLUMN * (long) id;
+    if (finalId < 0
+        || finalId > RESERVED_FIELD_IDS_START
+        || (finalId >= METADATA_SPACE_FIELD_ID_START
+            && idSpaceStart != METADATA_SPACE_FIELD_ID_START)) {
+      // ID overflows
+      return -1;
+    }
+
+    return (int) finalId;
+  }
+
+  public static int fieldIdFor(int statsFieldId) {
+    if (statsFieldId < 0 || statsFieldId % NUM_STATS_PER_COLUMN != 0) {
+      return -1;
+    }
+
+    int finalId;
+    if (statsFieldId < METADATA_SPACE_FIELD_ID_START) {
+      finalId = (statsFieldId - DATA_SPACE_FIELD_ID_START) / NUM_STATS_PER_COLUMN;
+    } else {
+      // this is a reserved field ID, which uses a different calculation
+      finalId =
+          statsFieldId
+              - RESERVED_FIELD_IDS
+              + (Integer.MAX_VALUE - statsFieldId)
+              + (statsFieldId - METADATA_SPACE_FIELD_ID_START) / NUM_STATS_PER_COLUMN;
+    }
+
+    return Math.max(-1, finalId);
+  }
+
+  public static Types.NestedField contentStatsFor(Schema schema) {
+    return TypeUtil.visit(schema, new ContentStatsSchemaVisitor());
+  }
+
+  private static Types.StructType contentStatsFor(Type type, int id) {
+    int fieldId = id;
+
+    return Types.StructType.of(
+        optional(fieldId++, "column_size", Types.LongType.get(), "Total size on disk"),
+        optional(
+            fieldId++,
+            "value_count",
+            Types.LongType.get(),
+            "Total value count, including null and NaN"),
+        optional(fieldId++, "nan_value_count", Types.LongType.get(), "Total NaN value count"),
+        optional(fieldId++, "null_value_count", Types.LongType.get(), "Total null value count"),
+        optional(fieldId++, "lower_bound", type, "Lower bound"),
+        optional(fieldId, "upper_bound", type, "Upper bound"));
+  }
+
+  private static class ContentStatsSchemaVisitor extends TypeUtil.SchemaVisitor<Types.NestedField> {
+    private final List<Types.NestedField> statsFields = Lists.newArrayList();
+
+    @Override
+    public Types.NestedField schema(Schema schema, Types.NestedField structResult) {
+      return optional(
+          146,
+          "content_stats",
+          Types.StructType.of(
+              statsFields.stream()
+                  .filter(Objects::nonNull)
+                  .sorted(Comparator.comparing(Types.NestedField::fieldId))
+                  .collect(Collectors.toList())));
+    }
+
+    @Override
+    public Types.NestedField list(Types.ListType list, Types.NestedField elementResult) {
+      list.fields()
```

Comment:
Like the below example, this just wraps ElementField in a list

Reply:
I guess it makes the code all look more similar ... so maybe it's good to keep these calls to field